src/rsa.rs

```diff
@@ -475,6 +486,51 @@ extern {
                                  rng: *mut rand::RAND) -> c::int;
 }
 
+#[cfg(feature = "rsa_signing")]
+#[allow(non_snake_case)]
+#[doc(hidden)]
+#[no_mangle]
+pub unsafe extern fn BN_rand_range(dest: *mut BIGNUM, range: *const BIGNUM,
+                                   rng: *mut rand::RAND) -> c::int {
+    const ERR: c::int = 0;
+    const SUCCESS: c::int = 1;
+
+    if (*range).neg != 0 {
+        return ERR;
+    }
+
+    let range = limb::Range {
+        limbs: core::slice::from_raw_parts((*range).d, (*range).top as usize),
+    };
+
+    if !range.valid().is_ok() {
+        return ERR;
+    }
+
+    let mut buf = &mut [0u8; limb::MAX_RANGE_BYTES][..range.bytes_count()];
```

Comment:
In general I went with stack-allocated buffers that are as big as the maximum key we want to use, even where using the heap might be more natural. Tracking down those maximums took some inference, and it's not obvious they'll never change, so I'm not sure this is the right strategy.